cdk/stack.py

```diff
@@ -0,0 +1,198 @@
+            name=names.DOWNLOAD_FUNCTION,
+            lambda_code_bucket = lambda_code_bucket,
+            timeout=900,
+            environment={
+                "ZOOM_VIDEOS_BUCKET": recordings_bucket.bucket.bucket_name,
+                "DOWNLOAD_QUEUE_NAME": queues.download_queue.queue_name,
+                "DEADLETTER_QUEUE_NAME": queues.download_dlq.queue.queue_name,
+                "UPLOAD_QUEUE_NAME": queues.upload_queue.queue_name,
+                "CLASS_SCHEDULE_TABLE": schedule.table.table_name,
+                "DEBUG": "0",
+                "ZOOM_ADMIN_ID": zoom_admin_id,
+                "ZOOM_API_KEY": zoom_api_key,
+                "ZOOM_API_SECRET": zoom_api_secret,
+                "LOCAL_TIME_ZONE": local_time_zone,
+                "DEFAULT_SERIES_ID": default_series_id,
+                "DOWNLOAD_MESSAGES_PER_INVOCATION": download_message_per_invocation,
+            }
+        )
+
+        # grant downloader function permissions
+        queues.download_queue.grant_consume_messages(downloader.function)
+        queues.upload_queue.grant_consume_messages(downloader.function)
+        schedule.table.grant_read_write_data(downloader.function)
+        recordings_bucket.bucket.grant_write(downloader.function)
+
+        op_counts = ZipOpCountsFunction(self, 'OpCountsFunction',
+            name=names.OP_COUNTS_FUNCTION,
+            lambda_code_bucket = lambda_code_bucket,
+            environment={}
+        )
+
+        # uploader lambda uploads recordings to opencast
+        uploader = ZipUploaderFunction(self, 'UploaderFunction',
+            name=names.UPLOAD_FUNCTION,
+            lambda_code_bucket=lambda_code_bucket,
+            timeout=300,
+            vpc_id=oc_vpc_id,
+            security_group_id=oc_security_group_id,
+            environment = {
+                "OPENCAST_API_USER": opencast_api_user,
+                "OPENCAST_API_PASSWORD": opencast_api_password,
+                "DEFAULT_PUBLISHER": default_publisher,
+                "OVERRIDE_PUBLISHER": override_publisher,
+                "OVERRIDE_CONTRIBUTOR": override_contributor,
+                "OC_WORKFLOW": oc_workflow,
+                "OC_FLAVOR": oc_flavor,
+                "OC_TRACK_UPLOAD_MAX": oc_track_upload_max,
+                "OPENCAST_BASE_URL": oc_base_url,
+                "ZOOM_VIDEOS_BUCKET": recordings_bucket.bucket.bucket_name,
+                "UPLOAD_QUEUE_NAME": queues.upload_queue.queue_name,
+                "DEBUG": "0",
+                "OC_OP_COUNT_FUNCTION": op_counts.function.function_name
+            }
+        )
+
+        # grant uploader function permissions
+        queues.upload_queue.grant_consume_messages(uploader.function)
+        op_counts.function.grant_invoke(uploader.function)
+        # this is required so that the presigned s3 urls generated by
+        # the uploader function will be valid
+        recordings_bucket.bucket.grant_read(uploader.function)
```

Comment:
since ffprobe check has been fixed so that it is actually called, we need `recordings_bucket.bucket.grant_read(downloader.function)` here